strawberry/directive.py

```diff
@@ -13,7 +13,7 @@
 from strawberry.field import StrawberryField
 
 
-def directive_field(name: str) -> StrawberryField:
```

Comment:
Why is this getting changed? This always return `StrawberryField`?

Reply:
Yes, but it conflicts with its use case. For example:

```python
import strawberry
from strawberry.schema_directive import Location


@strawberry.schema_directive(locations=[Location.OBJECT])
class SomeSchemaDirective:
    foo: str = strawberry.directive_field(name="foo")
    bar: int
```

If the typing says it returns a `StrawberryField`, that means I can't set it on a variable typed as anything else (i.e. in that example where the type expects `str`).

```
/tmp/xxx/xxx.py
  /tmp/xxx/xxx.py:7:16 - error: Expression of type "StrawberryField" cannot be assigned to declared type "str"
    "StrawberryField" is incompatible with "str" (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations
```

This is the same reason to why `dataclasses.field` is also typed as `Any` (it is actually overloaded, with the other possibility being the type of variable passed to `default` when it is given)

Reply:
@bellini666 we should implement this for directives: https://peps.python.org/pep-0681/

Reply:
oh, looks like you did already!

Reply:
@patrick91 I already did :)

https://github.com/strawberry-graphql/strawberry/pull/1723/files#diff-f2d5b9421987cb0ccccf8d32cad8245d96c5aaef62893bdd9e94e6aa8fa5c253R39